syntax = "proto3";

package Chatter;

service ServerServices {
    rpc Ping (PingRequest) returns (PingResponse);
}

service ClientServices {
    rpc Ping (PingRequest) returns (PingResponse);
}

message Empty {}

message PingRequest {
    string from = 1;
}

message PingResponse {
    string from = 1;
}

message ChatMessage {
    string sender = 1;
    string reciever = 2;
    enum MessageType {
        Message = 0;
        InviteRequest = 1;
    }
    string payload = 3;
    MessageType messagetype = 4;
}